src/test/java/com/checkmarx/flow/cucumber/integration/config_provider/remote_repo/ConfigProviderRemoteRepoSteps.java

```diff
@@ -0,0 +1,108 @@
+        switch (scanner) {
+            case ScaProperties.CONFIG_PREFIX:
+                scaConfiguration = configProvider.getConfiguration(uid, ScaProperties.CONFIG_PREFIX, ScaConfig.class);
+                break;
+            case AstProperties.CONFIG_PREFIX:
+                astConfiguration = configProvider.getConfiguration(uid, AstProperties.CONFIG_PREFIX, ASTConfig.class);
+                break;
+            default:
+                throw new MachinaRuntimeException("Scanner: " + scanner + " is not yet supported");
+        }
+    }
+
+    @Then("{string} configuration on Cx-Flow side should match the remote repo configuration data")
+    public void validateScannerConfigurationProperties(String scanner) {
+        switch (scanner) {
+            case ScaProperties.CONFIG_PREFIX:
+                assertScaConfiguration();
+                break;
+            case AstProperties.CONFIG_PREFIX:
+                assertAstConfiguration();
+                break;
+            default:
+                throw new MachinaRuntimeException("Scanner: " + scanner + " is not yet supported");
+        }
+
+    }
+
+    private void assertAstConfiguration() {
+        Assert.assertEquals("https://ast.astcheckmarx.com//config-provider-test", astConfiguration.getApiUrl());
+        Assert.assertEquals("default", astConfiguration.getPreset());
+        Assert.assertFalse(astConfiguration.isIncremental());
+        Assert.assertEquals(System.getenv("JAVA_HOME"), astConfiguration.getClientSecret());
+        Assert.assertEquals("astClientId", astConfiguration.getClientId());
+    }
+
+    private void assertScaConfiguration() {
+        Assert.assertEquals("https://sca.scacheckmarx.com//config-provider-test", scaConfiguration.getAppUrl());
+        Assert.assertEquals("https://api.scacheckmarx.com//config-provider-test", scaConfiguration.getApiUrl());
+        Assert.assertEquals("https://platform.checkmarx.net//config-provider-test", scaConfiguration.getAccessControlUrl());
+        Assert.assertEquals("cxflow", scaConfiguration.getTenant());
+        Assert.assertEquals("{LOW=25, MEDIUM=18, HIGH=15}", StringUtils.join(scaConfiguration.getThresholdsSeverity()));
+        Assert.assertEquals((Double)8.5, scaConfiguration.getThresholdsScore());
+        Assert.assertEquals("[high, medium, low]", scaConfiguration.getFilterSeverity().toString());
+        Assert.assertEquals((Double)7.5, scaConfiguration.getFilterScore());
+    }
+
+    private void init() {
+        String nameSpace = "cxflowtestuser";
+        String repoName = "configProviderTestRepo";
+        String branchName = "main";
+
+        try {
+            ConfigProvider configProvider = ConfigProvider.getInstance();
+            configProvider.init(uid, new RepoReader(gitHubProperties.getApiUrl(), nameSpace,
+                    repoName, branchName,
+                    gitHubProperties.getToken(), sourceProviderType));
+        } catch (ConfigurationException e) {
```

Comment:
There is no need to use try-catch here. If there is an exception, Cucumber will fail the test and display the exception details.

Reply:
I removed it.